tests/publiccloud/boottime.pm

```diff
@@ -0,0 +1,184 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Collect/store bootup time from publiccloud instances.
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+
+sub extract_startup_timings {
+    my $string = shift;
+    my $res    = {};
+    $string =~ s/Startup finished in\s*//;
+    $string =~ s/=(.+)$/+$1 (overall)/;
+    for my $time (split(/\s*\+\s*/, $string)) {
+        if ($time =~ /((\d{1,2})min\s*)?(\d{1,2}\.\d{1,3})s\s*\((\w+)\)/) {
+            my $sec = $3;
+            $sec += $2 * 60 if (defined($1));
+            $res->{$4} = $sec;
+        }
+    }
+    map { die("Fail to detect $_ timing") unless exists($res->{$_}) } qw(kernel initrd userspace overall);
+    return $res;
+}
+
+sub do_systemd_analyze {
+    my ($instance, %args) = @_;
+    $args{timeout} = 120;
+    my $start_time = time();
+    my $output     = "";
+
+    while ($output !~ /Startup finished in/ && time() - $start_time < $args{timeout}) {
+        $output = $instance->run_ssh_command(cmd => 'systemd-analyze', quiet => 1, proceed_on_failure => 1);
+        sleep 5;
+    }
+
+    die("Unable to get system-analyze in $args{timeout} seconds") unless (time() - $start_time < $args{timeout});
+    return extract_startup_timings($output);
+}
+
+sub build_influx_kv {
```

Comment:
this function requires some short explanation what and why happening here

Reply:
... or maybe just better name :) what about 
```suggestion
sub build_keyvalue_string {
```

Reply:
It is really influx related. I will add a comment and mention https://docs.influxdata.com/influxdb/v1.7/write_protocols/line_protocol_tutorial/#special-characters-and-keywords

Reply:
done